API/common/auth.py

```diff
@@ -66,24 +66,17 @@ def wrapper(*args, **kwargs) -> HttpResponse:
     return wrapper
 
 
-def _get_auth_token(scope):
-    client_id = os.environ.get("CLIENT_ID")
-    authority = os.environ.get("AUTHORITY_URL")
-    client_credential = os.environ.get("CLIENT_SECRET")
-    
-    if client_id is None or authority is None or client_credential is None:
-        logger.error("Missing required environment variables to get auth token!")
-        return None
-
-    app = msal.ConfidentialClientApplication(client_id, authority, client_credential)
+def _get_auth_token(*scopes: str):
     try:
-        result = app.acquire_token_for_client(scopes=scope)
-        return result['access_token']
+        credential = DefaultAzureCredential()
+        token = credential.get_token(*scopes)
+        token = "Bearer " + token.token
+        return token
```

Comment:
May I ask why do we need to modify the way of obtaining auth token?

Reply:
The previous code did not generate a valid token. This method of authentication is much simpler.